NDTensors/src/lib/TypeParameterAccessors/test/runtests.jl

```diff
@@ -4,188 +4,359 @@ using NDTensors.TypeParameterAccessors
 using NDTensors.TypeParameterAccessors: specify_parameters, ndims
 using LinearAlgebra: Transpose
 
-@testset "Test NDTensors.TypeParameterAccessors" begin
-  @testset "Get parameters" begin
-    @test @inferred(parameters(Array{Float32,3})) == (Float32, 3)
-    @test @inferred(parameters(Array{Float32,2}(undef, (2, 3)))) == (Float32, 2)
-    @test @inferred((() -> parameter(Array{Float32,3}, 1))()) == Float32
-    @test @inferred((() -> parameter(Array{Float32,3}, 2))()) == 3
-
-    @test @inferred(parameter(Array{Float32,3}, Position(1))) == Float32
-    @test @inferred(parameter(Array{Float32,3}, Position(2))) == 3
-    @test nparameters(Array) == 2
-    @test nparameters(Base.SubArray) == 5
-  end
+if VERSION ≥ v"1.8"
```

Comment:
Here's a suggestion for avoiding this code duplication across tests. We can define our own macro:
```julia
using Test

macro inferred_above(version, expr)
  quote
    if VERSION ≥ $(version)
      @inferred $(expr)
    else
      $(expr)
    end
  end
end

f(x) = x > 0.5 ? Int : Float64
@inferred_above v"1.7" f(0.2)
@inferred_above v"1.10" f(0.2)
```
which only tests inference if the version is at or above a certain value. So then we can use all of the same tests, and adjust the version as needed if we know inference fails on older versions but we don't care if that's the case.